src/directLine.ts

```diff
@@ -479,8 +483,10 @@ export class DirectLine implements IBotConnection {
     private tokenRefreshSubscription: Subscription;
 
     constructor(options: DirectLineOptions & Partial<Services>) {
+        this.acquireToken = options.acquireToken;
+        this.siteId = options.siteId;
         this.secret = options.secret;
-        this.token = options.secret || options.token;
+        this.token = this.acquireToken ? this.acquireToken() : options.secret || options.token;
```

Comment:
I'm concerned about the async nature of this function. What if the caller needs to do some HTTP call or database operation? I'm not sure this will work.

Reply:
Yeah, that doesn't look good. However, ABS API expects either a token or secret in the Authentication header. Maybe we should use a variable name that suits both.

Reply:
The client calls webchat after fetching the token atleast once. The WebSite code logic keeps refreshing the token. Yes, that code is async. Sample is in this [doc](https://microsoft-my.sharepoint-df.com/:w:/p/edwinb/EVNFyZJqlWhLoBiriD7KqHkBylNnPWAHKA83jl448nf6Ng?e=xVMBji).

Reply:
_this_ code doesn't look to be async, is what I mean. Can you write a test that exercises this to demonstrate the behavior you mention?